new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Installer.java

```diff
@@ -122,4 +137,45 @@ private String buildDownloadCommand() {
         params.fileName());
     return command;
   }
+
+  // Figure out the actual release number of the latest
+  static String getActualVersionIfNeeded(
+      String location,
+      String type,
+      String version
+  ) {
+    if (version == null || version.equalsIgnoreCase("latest")) {
+      String command = String.format(
+          "curl -fL %s/releases/latest -o %s/%s-%s", 
+          location,
+          DOWNLOAD_DIR,
+          type,
+          TMP_FILE_NAME);
+      
+      if (!Command.withParams(defaultCommandParams().command(command)).execute()) {
+        return null;
```

Comment:
What does it mean to return null here?  If the command fails isn't that a bigger issue?

Reply:
As I mentioned in the earlier responses, returning null is a way to tell the caller that this failed, and the caller will log a severe message, and return false to indicate a failure condition. The failure usually does not accompanied by an exception. We could manufacture an exception and throw here, but I am not sure what exception to throw.

Reply:
Keep in mind we are trying to make it clear for the "next" person.  If I understand this correctly, I think throwing an exception is clearer that "things are not working as expected".  Create a new exception class if it helps make things more clear.

Reply:
Agreed. working on a change.